tests/mocks.py

```diff
@@ -145,6 +147,19 @@ def revision(
             }
         )
 
+        next_free_txn_id = len(self._transactions)
+        for txn_id, comment in enumerate(comments, start=next_free_txn_id):
+            # See https://phabricator.services.mozilla.com/conduit/method/transaction.search/  # noqa
+            # for the Conduit API result object structure.
+            self._transactions.append(
+                {
+                    "id": txn_id,
+                    "type": "comment",
+                    "objectPHID": phid,
+                    "comments": [{"content": {"raw": comment}}],
+                }
```

Comment:
This is missing a bunch of fields that transactions have, which should be there even if filled with bogus data:
```json
{
      "id": 1003798,
      "phid": "PHID-XACT-DREV-3qlwutlsdcfwrge",
      "type": "subscribers",
      "authorPHID": "PHID-APPS-PhabricatorHeraldApplication",
      "objectPHID": "PHID-DREV-pm6r32ocou6ocjik4i5n",
      "dateCreated": 1559779750,
      "dateModified": 1559779750,
      "groupID": "zmgrbvzbcclaubtg4yt2ihwwotxewc4h",
      "comments": [],
      "fields": {
        "operations": [
          {
            "operation": "add",
            "phid": "PHID-USER-zp4gy3jjzfehipsktf6x"
          }
        ]
      }
```
It'd be good to run some `transaction.search`'s on production Phabricator and figure out what exactly is there.

You can take a look athow some of the other methods handle things like `dateCreated` etc.

Reply:
I've added a bunch.  See the new `transaction()` method.